cmd/api/discovery/apiresources.go

```diff
@@ -1,23 +1,36 @@
 package discovery
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/gin-gonic/gin"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/rest"
 
 	"github.com/kubearchive/kubearchive/cmd/api/abort"
+	"github.com/kubearchive/kubearchive/pkg/cache"
 )
 
+// Arbitrary time for cache, probably enough to not overload the Kubernetes API
+var cacheExpirationTime = 10 * time.Minute
+
 // GetAPIResource set apiResource attribute in the context based on the path parameters group version and resourceType
 //
 // This function relied on the DiscoveryClient's ServerResourcesForGroupVersion function. However it does not accept a context
 // so we had to implement the same call to be able to pass the context so telemetry traces are tied together
 // @rh-hemartin opened a ticket to allow for a context on that function, see https://github.com/kubernetes/client-go/issues/1370
-func GetAPIResource(client rest.Interface) gin.HandlerFunc {
+func GetAPIResource(client rest.Interface, cache *cache.Cache) gin.HandlerFunc {
```

Comment:
I would like to have explicitly set that we aren't using the [discoveryclient memcached](https://github.com/kubernetes/client-go/blob/master/discovery/cached/memory/memcache.go) because we just need the singular string of the resource and we can reuse the memory cache that we use for the other middleware.

Reply:
Done